File: src/backend/models/job.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class Job(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: Optional[int] = None
    session_id: str
    site: str
    title: str
    company: Optional[str] = None
    company_url: Optional[str] = None
    job_url: Optional[str] = None
    location_country: Optional[str] = None
    location_city: Optional[str] = None
    location_state: Optional[str] = None
    is_remote: bool = False
    description: Optional[str] = None
    tags: Optional[str] = None
    job_type: Optional[str] = None
    min_amount: Optional[float] = None
    max_amount: Optional[float] = None
    currency: Optional[str] = None
    interval: Optional[str] = None
    date_posted: Optional[datetime] = None
    scraped_at: datetime
    emails: Optional[str] = None
    job_level: Optional[str] = None
    company_industry: Optional[str] = None

File: src/renderer/src/types/job.ts
// types/job.ts
export interface Job {
  id: number;
  session_id: string;
  site: string;
  title: string;
  company: string | null;
  company_url: string | null;
  job_url: string;
  location_country: string | null;
  location_city: string | null;
  location_state: string | null;
  is_remote: boolean;
  description: string;
  tags: string;
  job_type: string;
  min_amount: number | null;
  max_amount: number | null;
  currency: string | null;
  interval: string | null;
  date_posted: string;
  scraped_at: string;
}
